eadk_discord/bot_setup.py

```diff
@@ -31,139 +28,153 @@ async def date_autocomplete(interaction: Interaction, current: str) -> list[Choi
-    async def book(
-        interaction: Interaction,
-        booking_date_arg: str | None,
-        user: Member | None,
-        desk_num_arg: Range[int, 1] | None,
-        end_date_arg: str | None,
-    ) -> None:
-        await eadk_bot.book(
-            CommandInfo.from_interaction(interaction),
-            booking_date_arg,
-            user.id if user else None,
-            desk_num_arg,
-            end_date_arg,
-        ).send(interaction)
-        database.save(database_path)
-
-    @bot.tree.command(name="unbook", description="Unbook a desk.", guilds=guilds)
-    @app_commands.autocomplete(booking_date_arg=date_autocomplete)
-    @app_commands.rename(booking_date_arg="date", desk_num_arg="desk_id", end_date_arg="end_date")
-    @app_commands.check(channel_check)
-    @app_commands.checks.has_any_role(TEST_SERVER_ROLE_ID, EADK_DESK_ADMIN_ID, EADK_DESK_REGULAR_ID)
-    async def unbook(
-        interaction: Interaction,
-        booking_date_arg: str | None,
-        user: Member | None,
-        desk_num_arg: Range[int, 1] | None,
-        end_date_arg: str | None,
-    ) -> None:
-        await eadk_bot.unbook(
-            CommandInfo.from_interaction(interaction),
-            booking_date_arg,
-            user.id if user else None,
-            desk_num_arg,
-            end_date_arg,
-        ).send(interaction)
-        database.save(database_path)
-
-    @bot.tree.command(
-        name="makeowned", description="Make a user the owner of the desk from a specific date onwards", guilds=guilds
-    )
-    @app_commands.autocomplete(start_date_str=date_autocomplete)
-    @app_commands.rename(start_date_str="start_date", desk="desk_id")
-    @app_commands.check(channel_check)
-    @app_commands.checks.has_any_role(TEST_SERVER_ROLE_ID, EADK_DESK_ADMIN_ID)
-    async def makeowned(
-        interaction: Interaction,
-        start_date_str: str,
-        user: Member | None,
-        desk: Range[int, 1],
-    ) -> None:
-        await eadk_bot.makeowned(
-            CommandInfo.from_interaction(interaction), start_date_str, user.id if user else None, desk
-        ).send(interaction)
-        database.save(database_path)
-
-    @bot.tree.command(
-        name="makeflex", description="Make a desk a flex desk from a specific date onwards", guilds=guilds
-    )
-    @app_commands.autocomplete(start_date_str=date_autocomplete)
-    @app_commands.rename(start_date_str="start_date", desk="desk_id")
-    @app_commands.check(channel_check)
-    @app_commands.checks.has_any_role(TEST_SERVER_ROLE_ID, EADK_DESK_ADMIN_ID)
-    async def makeflex(interaction: Interaction, start_date_str: str, desk: Range[int, 1]) -> None:
-        await eadk_bot.makeflex(CommandInfo.from_interaction(interaction), start_date_str, desk).send(interaction)
+class BotConfig(BaseModel):
+    bot_token: str
+    database_path: Path
+    guild_ids: Sequence[int | str]
+    channel_ids: Sequence[int | str]
```

Comment:
probs not a big problem here, but it's a common footgun that `var: Sequence[str] = "hello"` is valid. and ofc the same with `Sequence[int|str] = "hello"`